src/instructlab/system/info.py

```diff
@@ -158,9 +163,33 @@ def get_sysinfo() -> typing.Dict[str, typing.Any]:
     return d
 
 
+def categorize_and_print(sysinfo: typing.Dict[str, typing.Any]):
+    """Automatically categorize and print system information"""
+    categories: dict[str, list[tuple[str, typing.Any]]] = {}
+
+    for key, value in sysinfo.items():
+        if key.startswith(("sys", "os", "platform", "memory")):
+            category = "platform"
+        elif key.startswith("instructlab"):
+            category = "instructlab"
+        else:
+            category = key.split(".")[0]
+
+        if category not in categories:
+            categories[category] = []
+        categories[category].append((key, value))
+
+    for idx, (category, items) in enumerate(categories.items()):
+        if idx > 0:
+            print()
+        print(f"{category.capitalize()} Information:")
```

Comment:
This doesn't handle "InstructLab" correctly

llama_cpp_python maybe shouldn't be capitalized?

Perhaps you should have a dict of display names for the known names, and fall back to the capitalize methodology if a new one shows up.

And I would vote for leaving off "Information".

Reply:
```
$ ilab system info
Platform:
  sys.version: 3.11.9 (v3.11.9:xx, Apr  2 2024, 07:12:50) [Clang 13.0.0 (clang-)]
  sys.platform: xx
  os.name: xx
  platform.release: 1.1.1
  platform.machine: arm64
  platform.node: xx
  platform.python_version: 3.11.9
  platform.cpu_brand: Apple M3 Pro
  memory.total: 36.00 GB
  memory.available: 16.41 GB
  memory.used: 18.76 GB

InstructLab:
  instructlab.version: 0.19.0b2.dev0
  instructlab-dolomite.version: 0.1.1
  instructlab-eval.version: 0.2.0
  instructlab-quantize.version: 0.1.0
  instructlab-schema.version: 0.4.1
  instructlab-sdg.version: 0.3.0
  instructlab-training.version: 0.4.2

Torch:
  torch.version: 2.3.1
  torch.backends.cpu.capability: NO AVX
  torch.version.cuda: None
  torch.version.hip: None
  torch.cuda.available: False
  torch.backends.cuda.is_built: False
  torch.backends.mps.is_built: True
  torch.backends.mps.is_available: True

llama_cpp_python:
  llama_cpp_python.version: 0.2.79
  llama_cpp_python.supports_gpu_offload: True
```